shell/browser/api/electron_api_safe_storage.cc

```diff
@@ -31,12 +31,19 @@ void SetElectronCryptoReady(bool ready) {
 #endif
 
 bool IsEncryptionAvailable() {
+#if BUILDFLAG(IS_LINUX)
+  // Calling IsEncryptionAvailable() before the app is ready results in a crash
+  // on Linux.
+  // Refs: https://github.com/electron/electron/issues/32206.
+  if (!Browser::Get()->is_ready())
+    return false;
+#endif
   return OSCrypt::IsEncryptionAvailable();
 }
 
 v8::Local<v8::Value> EncryptString(v8::Isolate* isolate,
                                    const std::string& plaintext) {
-  if (!OSCrypt::IsEncryptionAvailable()) {
+  if (!IsEncryptionAvailable()) {
     gin_helper::ErrorThrower(isolate).ThrowError(
         "Error while decrypting the ciphertext provided to "
```

Comment:
This error message is confusing if the error is that the method was called before app.on('ready'). Can we throw a better error message in that case?

Reply:
Isn't changing the error message breaking? Will this change be backported to v15 even if we change the error message in this PR?

Reply:
Changing an error message is not a breaking change.

However, you should distinguish in the error message between the case that `OSCrypt::IsEncryptionAvailable` was called and returned false, and the case that it wasn't called at all (because the app wasn't ready). So it's not really changing an error message, it's adding a new error message.

Reply:
@nornagon done, PTAL!